network/dag/state.go

```diff
@@ -146,7 +144,17 @@ func (s *state) Add(ctx context.Context, transaction Transaction, payload []byte
 			return err
 		}
 
-		return s.notifyObservers(contextWithTX, transaction, payload)
+		// The double notification call makes sure that the node that created the TX gets the same amount of
+		// notifications as a node that received the TX from the network.
+		// Give a private transaction, both the receiver and creator get 2 notifications: 1 with payload and 1 without.
+		// The creator gets both from here while the receiver gets the notification with payload via WritePayload.
+		// For public transactions, all nodes get 2 notifications: 1 with and 1 without a payload.
+		if payload != nil {
```

Comment:
given the comments above, the order is off? Because now it's first added with payload, and then without payload?

Notifying twice here is quite obscure because it depends on specific behavior of the observers. If we can't express what we want in the current observer setup, I'd rather refactor it instead. Seems like you'd actually want to emit the events here, rather than from the publisher?